### PR TITLE
pythonPackages.httpcore: 0.12.0 -> 0.12.3

### PR DESCRIPTION
**Pull request by risicle**
###### Motivation for this change
Just a bump. Some new tests to disable due to network access.

Not done a **full** rebuild on macos as full evaluations are failing on master right now.

Reverse dependency `python38Packages.fastapi` already failing for me.
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
> Not done a full rebuild on macos as full evaluations are failing on master right now.

That must have happened recently. Do you have a log then I'll try to fix it when my darwin machine is free.

> Reverse dependency python38Packages.fastapi already failing for me.

Don't think I tested that on darwin when I updated it.

**Comment by fabaff** [comment]
I started to work on fixing the build issues of starlette which leads to the one of fastapi in https://github.com/NixOS/nixpkgs/pull/111079 but I'm only half way there.

**Comment by risicle** [comment]
Darwin evaluation failure:

```
error: while evaluating the attribute 'coq' at nixpkgs/pkgs/top-level/coq-packages.nix:124:3:
while evaluating 'recurseIntoAttrs' at nixpkgs/lib/attrsets.nix:502:5, called from nixpkgs/pkgs/top-level/coq-packages.nix:121:17:
while evaluating 'mapDerivationAttrset' at nixpkgs/lib/meta.nix:46:29, called from nixpkgs/pkgs/top-level/coq-packages.nix:121:35:
while evaluating 'mkCoqPackages' at nixpkgs/pkgs/top-level/coq-packages.nix:98:19, called from nixpkgs/pkgs/top-level/coq-packages.nix:118:22:
while evaluating 'filterPackages' at nixpkgs/pkgs/top-level/coq-packages.nix:72:24, called from nixpkgs/pkgs/top-level/coq-packages.nix:100:5:
while evaluating 'filterCoqPackages' at nixpkgs/pkgs/top-level/coq-packages.nix:75:23, called from nixpkgs/pkgs/top-level/coq-packages.nix:72:55:
while evaluating anonymous function at nixpkgs/pkgs/top-level/coq-packages.nix:77:22, called from nixpkgs/pkgs/top-level/coq-packages.nix:77:7:
while evaluating 'optional' at nixpkgs/lib/lists.nix:254:20, called from nixpkgs/pkgs/top-level/coq-packages.nix:78:11:
while evaluating the attribute 'meta.coqFilter' at nixpkgs/pkgs/stdenv/generic/make-derivation.nix:372:19:
while evaluating the attribute 'meta' at nixpkgs/pkgs/build-support/coq/default.nix:68:3:
while evaluating 'optionalAttrs' at nixpkgs/lib/attrsets.nix:349:25, called from nixpkgs/pkgs/build-support/coq/default.nix:73:5:
while evaluating anonymous function at nixpkgs/pkgs/build-support/coq/meta-fetch/default.nix:36:1, called from nixpkgs/pkgs/build-support/coq/default.nix:42:13:
while evaluating 'switch' at nixpkgs/pkgs/build-support/coq/extra-lib.nix:139:26, called from nixpkgs/pkgs/build-support/coq/meta-fetch/default.nix:37:1:
while evaluating 'switch-if' at nixpkgs/pkgs/build-support/coq/extra-lib.nix:111:18, called from nixpkgs/pkgs/build-support/coq/extra-lib.nix:144:5:
while evaluating 'findFirst' at nixpkgs/lib/lists.nix:199:5, called from nixpkgs/pkgs/build-support/coq/extra-lib.nix:111:22:
while evaluating 'combine' at nixpkgs/pkgs/build-support/coq/extra-lib.nix:141:21, called from nixpkgs/pkgs/build-support/coq/extra-lib.nix:144:34:
while evaluating 'opTruncate' at nixpkgs/pkgs/build-support/coq/extra-lib.nix:7:28, called from nixpkgs/pkgs/development/coq-modules/Cheerios/default.nix:7:23:
while evaluating 'versionAtLeast' at nixpkgs/lib/strings.nix:503:24, called from nixpkgs/pkgs/build-support/coq/extra-lib.nix:8:10:
while evaluating 'versionOlder' at nixpkgs/lib/strings.nix:491:22, called from nixpkgs/lib/strings.nix:503:29:
while evaluating 'truncate' at nixpkgs/pkgs/build-support/coq/extra-lib.nix:6:21, called from nixpkgs/pkgs/build-support/coq/extra-lib.nix:8:14:
while evaluating 'sublist' at nixpkgs/lib/lists.nix:589:5, called from nixpkgs/pkgs/build-support/coq/extra-lib.nix:6:46:
while evaluating the attribute 'coq-version' at nixpkgs/pkgs/applications/science/logic/coq/default.nix:75:12:
while evaluating anonymous function at nixpkgs/pkgs/build-support/coq/meta-fetch/default.nix:36:1, called from nixpkgs/pkgs/applications/science/logic/coq/default.nix:47:13:
while evaluating 'switch' at nixpkgs/pkgs/build-support/coq/extra-lib.nix:139:26, called from nixpkgs/pkgs/build-support/coq/meta-fetch/default.nix:37:1:
while evaluating 'switch-if' at nixpkgs/pkgs/build-support/coq/extra-lib.nix:111:18, called from nixpkgs/pkgs/build-support/coq/extra-lib.nix:144:5:
while evaluating 'findFirst' at nixpkgs/lib/lists.nix:199:5, called from nixpkgs/pkgs/build-support/coq/extra-lib.nix:111:22:
while evaluating 'combine' at nixpkgs/pkgs/build-support/coq/extra-lib.nix:141:21, called from nixpkgs/pkgs/build-support/coq/extra-lib.nix:144:34:
undefined variable 'isPath' at nixpkgs/pkgs/build-support/coq/meta-fetch/default.nix:63:12
```

**Comment by SuperSandro2000** [comment]
> Not done a full rebuild on macos as full evaluations are failing on master right now.

That must have happened recently. Do you have a log then I'll try to fix it when my darwin machine is free.

> Reverse dependency python38Packages.fastapi already failing for me.

Don't think I tested that on darwin when I updated it.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 111348` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
14 packages marked as broken and skipped:
</summary>
<ul>
<li>python37Packages.fastapi</li>
<li>python38Packages.fastapi</li>
<li>python39Packages.PyRMVtransport</li>
<li>python39Packages.axis</li>
<li>python39Packages.datasette</li>
<li>python39Packages.entrance</li>
<li>python39Packages.entrance-with-router-features</li>
<li>python39Packages.fastapi</li>
<li>python39Packages.google-music</li>
<li>python39Packages.httpcore</li>
<li>python39Packages.httpx</li>
<li>python39Packages.respx</li>
<li>python39Packages.sanic</li>
<li>python39Packages.sanic-auth</li>
</ul>
</details>
<details>

  <summary>4 packages failed to build and are new build failures:</summary>
  <ul>
<li>python38Packages.entrance: log was empty</li>
<li>python38Packages.entrance-with-router-features: log was empty</li>
<li>python38Packages.sanic: log <a href="https://termbin.com/ovt8q">https://termbin.com/ovt8q</a></li>
<li>python38Packages.sanic-auth: log was empty</li>
  </ul>
</details>
<details>
<summary>
20 packages built:
</summary>
<ul>
<li>datasette (python38Packages.datasette)</li>
<li>google-music-scripts</li>
<li>home-assistant</li>
<li>python37Packages.PyRMVtransport</li>
<li>python37Packages.axis</li>
<li>python37Packages.datasette</li>
<li>python37Packages.entrance</li>
<li>python37Packages.entrance-with-router-features</li>
<li>python37Packages.google-music</li>
<li>python37Packages.httpcore</li>
<li>python37Packages.httpx</li>
<li>python37Packages.respx</li>
<li>python37Packages.sanic</li>
<li>python37Packages.sanic-auth</li>
<li>python38Packages.PyRMVtransport</li>
<li>python38Packages.axis</li>
<li>python38Packages.google-music</li>
<li>python38Packages.httpcore</li>
<li>python38Packages.httpx</li>
<li>python38Packages.respx</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 111348` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>1 package failed to build and are new build failure:</summary>
  <ul>
<li>python38Packages.fastapi: log was empty</li>
  </ul>
</details>
<details>
<summary>
25 packages built:
</summary>
<ul>
<li>datasette (python38Packages.datasette)</li>
<li>google-music-scripts</li>
<li>home-assistant</li>
<li>python37Packages.PyRMVtransport</li>
<li>python37Packages.axis</li>
<li>python37Packages.datasette</li>
<li>python37Packages.entrance</li>
<li>python37Packages.entrance-with-router-features</li>
<li>python37Packages.fastapi</li>
<li>python37Packages.google-music</li>
<li>python37Packages.httpcore</li>
<li>python37Packages.httpx</li>
<li>python37Packages.respx</li>
<li>python37Packages.sanic</li>
<li>python37Packages.sanic-auth</li>
<li>python38Packages.PyRMVtransport</li>
<li>python38Packages.axis</li>
<li>python38Packages.entrance</li>
<li>python38Packages.entrance-with-router-features</li>
<li>python38Packages.google-music</li>
<li>python38Packages.httpcore</li>
<li>python38Packages.httpx</li>
<li>python38Packages.respx</li>
<li>python38Packages.sanic</li>
<li>python38Packages.sanic-auth</li>
</ul>
</details>

**Review by mweinelt — APPROVED** [review]
